sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -351,11 +351,12 @@ class ParquetFileFormat
       if (pushed.isDefined) {
         ParquetInputFormat.setFilterPredicate(hadoopAttemptContext.getConfiguration, pushed.get)
       }
+      val taskContext = Option(TaskContext.get())
```

Comment:
If we require the `TaskContext` must exist here, then we can remove the `forceUseOnHeap` parameter in `initBatch`?

Reply:
Why can't we make this assumption?